public/app/features/query/state/runRequest.ts

```diff
@@ -194,7 +194,7 @@ export function callQueryMethod(
   );
 
   // If its a public datasource, just return the result. Expressions will be handled on the backend.
-  if (datasource.type === 'public-ds') {
+  if (!!config.publicDashboardAccessToken) {
```

Comment:
```suggestion
  if (config.publicDashboardAccessToken != null) {
```
maye 🤷🏻 -- supports both null and undefined

Reply:
```suggestion
  if (config.publicDashboardAccessToken) {
```

Why does this not work? And empty string is false

Reply:
I put !! in order to make it a boolean value, but it's the same when it's inside an if. I can change it